docs/source/configure.rst

```diff
@@ -22,90 +15,92 @@ without passing parameters the ``config.json`` will be used to populate the valu
+    max_download_attempts = 3
 
-$ sentinelhub.config --instance_id <your instance id>
-$ sentinelhub.config --sh_client_id <your client id> --sh_client_secret <your client secret>
+    [custom-profile]
+    instance_id = "something-else"
 
-or set them up by configuring an instance of :class:`~sentinelhub.config.SHConfig`:
+The file can also be updated programmatically by using the ``save`` method.
 
 .. code-block:: python
 
     from sentinelhub import SHConfig
 
     config = SHConfig()
+    config.instance_id = "my-instance-id"
+    config.save("my-profile")
 
-    config.instance_id = '<your instance id>'
-    config.sh_client_id = '<your client id>'
-    config.sh_client_secret = '<your client secret>'
 
+Another option is to update the configuration file via CLI. However this approach can only modify existing profiles, any new profiles need to be added manually or through the Python interface.::
 
-One can save these into the package ``config.json`` file by calling:
+$ sentinelhub.config --profile my-profile --instance_id my-instance-id
 
-.. code-block:: python
+Environment Variables
+*********************
 
-    config.save()
+We generally suggest using the configuration file, but we offer limited support for environmental variables to simplify situations such as building docker images.
 
-Once set (either with command line or as described above), the default parameters to interact with Sentinel Hub
-will be read from ``config.json``, unless you purposely specify an instance of :class:`~sentinelhub.config.SHConfig`
-object containing different parameters.
+The ``SHConfig`` class reads the following environmental variables during initialization:
 
-.. admonition:: Additional information on creating OAuth client
+- ``SH_PROFILE`` that dictates which profile is loaded if not explicitly provided.
+- ``SH_CLIENT_ID`` and ``SH_CLIENT_SECRET`` for setting the SentinelHub credentials.
 
-    For detailed instructions on how to obtain credentials, you can see the `Sentinel Hub webinar`_.
 
+Precedence
+**********
 
+The general precedence order is ``explicit parameters > environment > configuration file > defaults``.
 
-Amazon S3 Configuration
-***********************
+This means that ``SHConfig(profile="my-profile", sh_client_id="my-id")`` will be taken into account over ``SH_PROFILE`` and ``SH_CLIENT_ID`` environment variables, which would take precedence over what is specified in the ``configuration.toml``.
 
-The package enables downloading Sentinel-2 L1C and L2A data from `Amazon S3`_ storage buckets. The data is contained in
-Requester Pays buckets therefore `AWS credentials`_ are required to use these capabilities. The credentials
-can be set in the package configuration file with parameters ``aws_access_key_id`` and ``aws_secret_access_key``. This
-can be configured using the command line as::
 
-$ sentinelhub.config --aws_access_key_id <your access key> --aws_secret_access_key <your secret access key>
+Sentinel Hub Configuration
```

Comment:
I feel like this should be moved up, because it describes a more general overview of the configuration and offers some explanations. Perhaps at the top, even?

The AWS configuration could stay below, since it's a specific use case

Reply:
but this is only relevant to three of the parameters no?

Reply:
you're correct regarding the parameters POV, but the rest of the text reads like an intro which gives a broader meaning to the whole point of configuration. 

I'm not married to the idea, so if you don't like it, feel free to ignore.